### PR TITLE
fix(ui): adds relationTo to cellComponentProps

### PR DESCRIPTION
**Pull request by kendelljoseph**
## Description

Corrects issue where relation related cell data was not being displayed

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/ui/src/providers/ComponentMap/buildComponentMap/mapFields.tsx [hunk 1]
```diff
@@ -200,6 +200,7 @@ export const mapFields = (args: {
               : undefined,
           labels: 'labels' in field ? field.labels : undefined,
           options: 'options' in field ? field.options : undefined,
+          relationTo: 'relationTo' in field ? field.relationTo : undefined,
```
Comment (jacobsfletch) [inline packages/ui/src/providers/ComponentMap/buildComponentMap/mapFields.tsx:203]: This change should actually be placed in the `relationship` case below. This way no `in` operator needs to be used. The `in` operator is not type safe.

Comment (jacobsfletch) [reply packages/ui/src/providers/ComponentMap/buildComponentMap/mapFields.tsx:203]: Same with the `upload` case.

Comment (kendelljoseph) [reply packages/ui/src/providers/ComponentMap/buildComponentMap/mapFields.tsx:203]: `field.relationTo` is already already in those places.

Without also being in this location as well the cell doesn't populate the information for `upload`.

Try this test, with, and without this line added.

https://github.com/payloadcms/payload/blob/916923071a2c28077326bfd2059d1665ef47e94f/test/uploads/e2e.spec.ts#L73

Comment (jacobsfletch) [reply packages/ui/src/providers/ComponentMap/buildComponentMap/mapFields.tsx:203]: No those are being set on `fieldComponentProps` not `cellComponentProps`. That's the difference here. So just do `cellComponentProps.relationTo = field.relationTo`

**Review by jacobsfletch — APPROVED** [review]

**Pull request merged**